specification/requirements.adoc

```diff
@@ -1,5 +1,167 @@
 [[requirements]]
 == Requirements
 
-Hardware (platform and device) and software (host and guest) requirements to
-support CoVE-IO.
+In order to extend TVM TCBs with external and untrusted devices, the host
+platform and software stack, together with assigned devices, must meet a
+specific set of requirements.
+
+=== Device
+
+CoVE-IO compliant implementations support extending TVMs TCBs with devices if
+and only if they meet the following requirements:
+
+==== PCI
+
+A CoVE-IO supported device must be a PCIe device with optional SR-IOV
+capabilities. When supporting SR-IOV, the device can be split into multiple
+Virtual Functions (VFs). A TVM assigned interface can be the whole device, also
+known as the Physiscal Device (PF) or any of its VFs. In both cases, this unit
+of assignment is referred to as a TEE I/O Device Interface (TDI).
+
+A host enumerated TDI must provide the following PCIe capabilities:
+
+===== TEE Device Interface Protocol (TDISP)
+
+In order to interoperate with a RISC-V CoVE-IO implementation, PCIe devices must
+support the <<TDISP>> protocol, as defined in the latest TDISP ECN.
+
+In particular, CoVE-IO compatible device firmwares must run a Device Security
+Manager (DSM) to enforce the TDISP defined security attributes and policies.
+The DSM must support both full PCIe devices (PFs) and Virtual Functions (VFs)
+assignment to TVMs.
+
+TODO: List of required TDISP messages
+
+====== Enumeration
+
+TDIs PCIe Device Capabilities Registers must have the TEE-IO bit set in order
+for the host to discover their TDISP capability.
+
+===== Data Object Exchange (DOE)
+
+The TDI implements the optional PCIe Data Object Exchange mechanism. The minimal
+supported version is 1.0.
+
+A CoVE-IO supported device must support the following DOE object types:
+
+|===
+| Vendor ID | Object Type | Description
+
+| 0001 | 00 | DOE Discovery
+| 0001 | 01 | CMA/SPDM
+| 0001 | 02 | Secure CMA/SPDM
+|===
+
+====== Secure Protocol and Data Model (SPDM)
+
+RISC-V CoVE-IO compliant hosts use the Secure Protocol and Data Model (<<SPDM>>)
+protocol to exchange TDISP and IDE Key Management (IDE_KM) messages with
+physical devices, over DOE mailboxes. CoVE-IO also relies on SPDM for
+gathering devices certificates and measurements.
+
+As a consequence, a CoVE-IO supported DSM must support the SPDM specification
+version 1.2. It must also support sending and receiving SPDM Secured Messages
+(<<SecuredSPDM>>) version 1.1.
+
+TODO: List of required SPDM messages
```

Comment:
I think we beyond message, we may need to clarify the SPDM capability, such as CERT_CAP, ENCRYPT_CAP, MAC_CAP, KEY_EX_CAP.

Also, SPDM algorithms - the basic algorithm is defined in PCI CMA 1.0 ECN. It should be updated to new version to include the one for key exchange.